subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultVariantTransformRegistration.java

```diff
@@ -55,18 +57,23 @@
         hasher.putHash(classLoaderHierarchyHasher.getClassLoaderHash(implementation.getClassLoader()));
 
         // TODO - should snapshot later?
```

Comment:
This is the TODO I was referring to. I think it would be the right thing to snapshot lazily.